buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -650,6 +664,24 @@ public ByteBuf setZero(int index, int length) {
     }
 
     @Override
+    public int setCharSequence(int index, CharSequence sequence, Charset charset) {
+        if (charset.equals(CharsetUtil.UTF_8)) {
+            int len = sequence.length();
```

Comment:
it seems like the majority of the time the encoding will be `utf8` or `ascii`. what about adding methods to handle these cases on the `ByteBuf` interface? This could save the conditional checks every time and we could call these method here too.

```java
public abstract int setUtf8(int index, CharSequence seq);
public abstract int setAscii(int index, CharSequence seq);
public abstract int setCharSequence(int index, CharSequence sequence, Charset charset);
...
```